src/streamdatasetjson/datasetjson.py

```diff
@@ -1,22 +1,46 @@
+from typing import List
+import ijson
 import utils
 
 from dataset import Dataset
 from errors import DatasetNotFoundError
 
 
 class DatasetJSON:
+    """
+    Dataset JSON instance. 
+    """
     def __enter__(self):
         self._file = open(self._path, "r")
+        self._dataset_prefixes = self._generate_dataset_prefixes()
         return self
 
     def __exit__(self, type, value, traceback):
```

Comment:
Typing can be found here:
https://stackoverflow.com/questions/58808055/what-are-the-python-builtin-exit-argument-types